src/script/script_instance.cpp

```diff
@@ -364,8 +364,8 @@ static const SaveLoad _script_byte[] = {
 			SQInteger res;
 			sq_getinteger(vm, index, &res);
 			if (!test) {
-				int value = (int)res;
-				SlCopy(&value, 1, SLE_INT32);
+				int64 value = (int64)res;
```

Comment:
I think the cast should not be needed as SQInteger is already defined at `__int64`. Explicitly having a separate variable of the right type is definitely a good thing.